hub-detect/src/main/groovy/com/blackducksoftware/integration/hub/detect/tool/bazel/BazelExecutableFinder.java

```diff
@@ -0,0 +1,58 @@
+/**
+ * hub-detect
+ *
+ * Copyright (C) 2019 Black Duck Software, Inc.
+ * http://www.blackducksoftware.com/
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.blackducksoftware.integration.hub.detect.tool.bazel;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.blackducksoftware.integration.hub.detect.configuration.DetectConfiguration;
+import com.blackducksoftware.integration.hub.detect.configuration.DetectProperty;
+import com.blackducksoftware.integration.hub.detect.configuration.PropertyAuthority;
+import com.blackducksoftware.integration.hub.detect.detector.DetectorEnvironment;
+import com.blackducksoftware.integration.hub.detect.type.ExecutableType;
+import com.blackducksoftware.integration.hub.detect.util.executable.ExecutableFinder;
+
+public class BazelExecutableFinder {
+    private final Logger logger = LoggerFactory.getLogger(this.getClass());
+
+    private final ExecutableFinder executableFinder;
+    private final DetectConfiguration detectConfiguration;
+
+    private boolean hasLookedForSystemBazel = false;
+    private String resolvedBazel = null;
+
+    public BazelExecutableFinder(final ExecutableFinder executableFinder, final DetectConfiguration detectConfiguration) {
+        this.executableFinder = executableFinder;
+        this.detectConfiguration = detectConfiguration;
+    }
+
+    public String findBazel(final DetectorEnvironment environment) {
+        if (!hasLookedForSystemBazel) {
+            final String userProvidedBazelPath = detectConfiguration.getProperty(DetectProperty.DETECT_BAZEL_PATH, PropertyAuthority.None);
+            resolvedBazel = executableFinder.getExecutablePathOrOverride(ExecutableType.BAZEL, true, environment.getDirectory(), userProvidedBazelPath);
+                hasLookedForSystemBazel = true;
```

Comment:
Looks like it might fit the CacheableExecutableFinder model - might not need it's own class. But this is perfectly fine.